clients/go/pkg/zbc/client.go

```diff
@@ -189,25 +218,37 @@ func setDefaultCredentialsProvider(config *ClientConfig) error {
 	return nil
 }
 
-func configureConnectionSecurity(config *ClientConfig, opts *[]grpc.DialOption) error {
+func configureConnectionSecurity(config *ClientConfig, opts []grpc.DialOption) ([]grpc.DialOption, error) {
 	if !config.UsePlaintextConnection {
 		var creds credentials.TransportCredentials
 
 		if config.CaCertificatePath == "" {
 			creds = credentials.NewTLS(&tls.Config{})
 		} else if _, err := os.Stat(config.CaCertificatePath); os.IsNotExist(err) {
-			return fmt.Errorf("expected to find CA certificate but no such file at '%s': %w", config.CaCertificatePath, ErrFileNotFound)
+			return nil, fmt.Errorf("expected to find CA certificate but no such file at '%s': %w", config.CaCertificatePath, ErrFileNotFound)
 		} else {
 			creds, err = credentials.NewClientTLSFromFile(config.CaCertificatePath, "")
 			if err != nil {
-				return err
+				return nil, err
 			}
 		}
 
-		*opts = append(*opts, grpc.WithTransportCredentials(creds))
+		opts = append(opts, grpc.WithTransportCredentials(creds))
 	} else {
-		*opts = append(*opts, grpc.WithInsecure())
+		opts = append(opts, grpc.WithInsecure())
 	}
 
-	return nil
+	return opts, nil
+}
+
+func configureKeepAlive(config *ClientConfig, opts []grpc.DialOption) ([]grpc.DialOption, error) {
+	keepAlive := DefaultKeepAlive
+
+	if config.KeepAlive < time.Duration(0) {
+		return nil, errors.New("Keep alive must be a positive duration")
+	} else if config.KeepAlive != time.Duration(0) {
+		keepAlive = config.KeepAlive
+	}
+
+	return append(opts, grpc.WithKeepaliveParams(keepalive.ClientParameters{Time: keepAlive, PermitWithoutStream: true})), nil
```

Comment:
Remind me again why we want to permit without stream here? Isn't that allowing keep alives on a connection even if there are no outstanding RPCs?

Reply:
We don't, actually. I think I inverted that when testing and forgot to revert, my bad. Thanks for catching that.